### DISCUSSION TITLE
Saving custom data to Google Sheets

### DISCUSSION THREAD

**Discussion by doroness** [AutomaApp/automa#332] [Q&A] [answered]
Trying to save data {activeTabUrl} to a google sheet., No matter how tried to format it, the process produces an error when it comes to that stage. Tried several formats:

```
[
  { "URL": activeTabUrl}
]

[ 
  ["URL"], [ activeTabUrl] 
]

["URL",activeTabUrl]

["URL" , {{activeTabUrl}} ]

[["URL"] , [ "URL",{{activeTabUrl}}]]

[["URL"],[ "URL",activeTabUrl]]
```

What am I doing wrong? I know the connection to the sheet is ok because 1) I get data from that sheet 2) when I use the "data from - Data Columns" option - it works OK.

[You can see my workflow here](https://www.automa.site/workflow/vuOAeZE3xkGYfH1-d7jmx)

<img width="1313" alt="image" src="https://user-images.githubusercontent.com/53006624/153687635-9cf73386-b182-49b0-95ba-1e33513bbe6e.png">

**Comment by Kholid060** [comment]
Try
`[["URL"], ["URL", "{{activeTabUrl}}"]]`

**Comment by doroness** [comment]
Hi
Thanks for your answer. It didn't work :( Any other Ideas? Is there something I can do to help you debug this?

**Comment by Kholid060** [comment, marked as answer]
Automa sending the right payload (first image), and I think the error is on the range that you inputted (second image) when trying to update the spreadsheet cells.

![image](https://user-images.githubusercontent.com/22908993/153730830-b69345ae-acc1-4bb9-b09d-e363c1cdf72f.png)
![image](https://user-images.githubusercontent.com/22908993/153730838-9d8ffe22-05cb-4ab2-867c-462210bcd2bb.png)

Because with `fullurl!A:A` range you requested to write the cells on the first column (A) but in the payload, it writes to the second column (B)
<pre>
[
  ["URL"],
  <i>["URL","{{activeTabUrl}}"]</i>
]
</pre>

There're two ways you can fix this, first by changing the range from `fullurl!A:A` to `fullurl!A:B`. And second, if you planning to only write on the first column, change the payload into

<pre>
[
  ["URL"],
  ["{{activeTabUrl}}"]
]
</pre>

**Reply by doroness** [reply]
Yep - that was it - it worked. Thanks!

How can I prevent Automa to keep rewriting the values on the same row in the sheet? I want to go down a line with every new update 

<img width="311" alt="image" src="https://user-images.githubusercontent.com/53006624/153732080-4434dd88-7fe5-46b2-a2a8-5d4067d4f656.png">

**Resolved - I used the loopData to set ranges dynamically.** 

Thank you very much!

**Comment by Kholid060** [comment]
And if you want to debug the network request, you can open `chrome://extensions`, find Automa and click the background page.
